libvast/src/detail/posix.cpp

```diff
@@ -318,26 +330,28 @@ caf::error make_blocking(int fd) {
   return make_nonblocking(fd, false);
 }
 
-caf::error poll(int fd, int usec) {
+caf::expected<bool> rpoll(int fd, int usec) {
   fd_set rdset;
```

Comment:
Can you call this `read_set` so that the implementation is symmetric to the `write_set` you introduced below?